samples-lib/src/main/java/com/esri/arcgismaps/sample/sampleslib/DownloaderActivity.kt

```diff
@@ -224,59 +231,56 @@ abstract class DownloaderActivity : AppCompatActivity() {
-                // create file at location to write the PortalItem ByteArray
-                provisionFile.createNewFile()
-                // create and write the file output stream
-                val writeOutputStream = FileOutputStream(provisionFile)
-                writeOutputStream.write(byteArray)
-
+            }.onSuccess {
                 // unzip the file if it is a .zip
                 if (portalItem.name.contains(".zip")) {
-                    val fileInputStream = FileInputStream(destinationFilePath)
-                    val zipInputStream = ZipInputStream(BufferedInputStream(fileInputStream))
-                    var zipEntry: ZipEntry? = zipInputStream.nextEntry
-                    val buffer = ByteArray(1024)
-                    while (zipEntry != null) {
-                        if(zipEntry.isDirectory){
-                            File(provisionLocation.path, zipEntry.name).mkdirs()
-                        }
-                        else {
-                            val file = File(provisionLocation.path, zipEntry.name)
-                            val fout = FileOutputStream(file)
-                            var count = zipInputStream.read(buffer)
-                            while (count != -1) {
-                                fout.write(buffer, 0, count)
-                                count = zipInputStream.read(buffer)
+                    // set up the file at the download path
+                    val provisionFile = File(destinationFilePath)
+                    // set up the input streams
+                    FileInputStream(destinationFilePath).use { fileInputStream ->
+                        ZipInputStream(BufferedInputStream(fileInputStream)).use { zipInputStream ->
+                            var zipEntry: ZipEntry? = zipInputStream.nextEntry
+                            val buffer = ByteArray(1024)
+                            while (zipEntry != null) {
+                                if (zipEntry.isDirectory) {
+                                    File(provisionLocation.path, zipEntry.name).mkdirs()
+                                } else {
+                                    val file = File(provisionLocation.path, zipEntry.name)
+                                    val fout = FileOutputStream(file)
+                                    var count = zipInputStream.read(buffer)
+                                    while (count != -1) {
+                                        fout.write(buffer, 0, count)
+                                        count = zipInputStream.read(buffer)
+                                    }
+                                    fout.close()
+                                }
+
+                                zipInputStream.closeEntry()
+                                zipEntry = zipInputStream.nextEntry
                             }
-                            fout.close()
+                            zipInputStream.close()
```

Comment:
This call should not be necessary, since the `use { }` block will take care of it.